### DISCUSSION TITLE
How to fix WebSockets so that it works on things that aren't only Browsers

### DISCUSSION THREAD

**Discussion by karl-police** [mitmproxy/mitmproxy#6455] [Q&A]
I ran mitmweb, then I did ``CTRL + SHIFT + R`` and then it started logging stuff.

But where can I see the messages within a WebSocket?

I also get this issue, even though I installed the certificate at the place where it said "Root" thing.
```
Unable to establish TLS connection with server (Certificate verify failed: unable to get local issuer certificate)
```

The WebSocket stuff works fine if it's from a browser though 
![image](https://github.com/mitmproxy/mitmproxy/assets/12023782/1748a5e0-a539-4db2-824e-e5edff89e10a)

**Comment by mhils** [comment]
To view WebSocket messages, select the WebSocket flow and then open the "Messages" tab on the left. :)

> Unable to establish TLS connection with server (Certificate verify failed: unable to get local issuer certificate)

This means mitmproxy does not trust the server's certificate. The two options to fix that are

1. https://docs.mitmproxy.org/stable/concepts-options/#ssl_insecure
2. https://docs.mitmproxy.org/stable/concepts-options/#ssl_verify_upstream_trusted_ca

**Reply by karl-police** [reply]
is ssl_insecure going in HTTPS or just straight goes to HTTP?

nvm, I think it's not forcing HTTP

**Reply by karl-police** [reply]
idk which certificate

I tried doing ``curl --proxy 127.0.0.1:8080 https://example.com`` aslong it's ``https`` it didn't work and it said "Unknown CA". Can't use self-signed CAs. Not sure which CA or whatever it is, I do not know. Someone please tell me.
